### ISSUE TITLE
Compile C++ indexer with Bazel using clang-3.5. [phab:D191]

### ISSUE THREAD

**Issue by kythe** [google/kythe#403]
_Created by **zarko** at 2015-04-16 00:36:37_: 

We might need to do some work on the CROSSTOOL file;
please let me know if this causes you problems with
your personal configuration.

These BUILD files will work only if you have LLVM
checked out to the default location. In future CLs we
should restore the configurability we have in Campfire.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-16 15:42:26_:

This file is basically Kythe's now.  We should move this into //tools/ and clean it up.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-16 15:42:26_:

Modifications should be listed in README.google.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-16 15:42:26_:

Modifications should be listed in README.google.

**Comment by kythe** [comment]
_**schroederc** required changes at 2015-04-16 15:42:26_

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-04-16 15:51:57_:

I also get this error when trying to run `bazel test //...`.  I'm assuming this is because I don't have exactly llvm-3.5 installed on my machine (3.4 instead).  This seems to be too harsh of a build constraint for both us and our users, though.

```
ERROR: /usr/local/google/home/schroederc/kythe/third_party/googleflags/BUILD:5:1: undeclared inclusion(s) in rule '//third_party/googleflags:gflags':
this rule is missing dependency declarations for the following files included by 'third_party/googleflags/src/gflags_reporting.cc':
  '/usr/lib/clang/3.4/include/stddef.h'
  '/usr/lib/clang/3.4/include/stdarg.h'
  '/usr/lib/clang/3.4/include/stdint.h'.
INFO: Elapsed time: 20.786s, Critical Path: 15.32s

Executed 0 out of 32 tests: 32 fail to build.
```

**Comment by kythe** [comment]
_**zarko** wrote at 2015-04-16 17:59:07_:

  - Autoconfigure header search paths.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-16 18:10:08_:

We should keep our files out of third_party and put this in tools/cpp/CROSSTOOL.

```
rm tools/cpp
mkdir tools/cpp
ln -s ../../third_party/bazel/tools/cpp/BUILD tools/cpp/
ln -s ../../third_party/bazel/tools/cpp/empty.cc tools/cpp/
ln -s ../../third_party/bazel/tools/cpp/osx_gcc_wrapper.sh tools/cpp/
```

**Comment by kythe** [comment]
_**schroederc** required changes at 2015-04-16 18:10:08_

**Comment by kythe** [comment]
_**zarko** wrote at 2015-04-16 18:21:26_:

  - Move files and make symlinks

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-04-16 18:31:44_:

third-party header?

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-04-16 18:31:44_:

I'm still getting the same error as before.

```
bazel test //...                                                                                                            ~/kythe [─✗─]
INFO: Found 196 targets and 32 test targets...
ERROR: /usr/local/google/home/schroederc/kythe/third_party/zlib/BUILD:5:1: undeclared inclusion(s) in rule '//third_party/zlib:zlib':
this rule is missing dependency declarations for the following files included by 'third_party/zlib/compress.c':
  '/usr/lib/clang/3.4/include/limits.h'
  '/usr/lib/clang/3.4/include/stddef.h'
  '/usr/lib/clang/3.4/include/stdarg.h'.
INFO: Elapsed time: 1.296s, Critical Path: 0.32s

Executed 0 out of 32 tests: 32 fail to build.
```

`./setup_bazel.sh` just says that it found clang at `/usr/bin/clang`.  `tools/cpp/CROSSTOOL` contains
```
  cxx_builtin_include_directory: "/usr/include/c++/4.8"
  cxx_builtin_include_directory: "/usr/include/c++/4.8/backward"
  cxx_builtin_include_directory: "/usr/include/x86_64-linux-gnu/c++/4.8"
  cxx_builtin_include_directory: "/usr/local/include"
  cxx_builtin_include_directory: "/usr/lib/llvm-3.4/lib/clang/3.4/include"
  cxx_builtin_include_directory: "/usr/lib/gcc/x86_64-linux-gnu/4.8/include"
  cxx_builtin_include_directory: "/usr/include/x86_64-linux-gnu"
  cxx_builtin_include_directory: "/usr/include"
```

**Comment by kythe** [comment]
_**schroederc** required changes at 2015-04-16 18:31:51_

**Comment by kythe** [comment]
_**zarko** wrote at 2015-04-16 19:56:31_:

  - Use realpath too.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-04-16 20:01:26_:

I'm getting some `-Wimplicit-function-declaration` and `-Wdeprecated-register` warnings now, but everything seems to build at least.

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2015-04-16 20:01:26_

**Comment by kythe** [comment]
_**zarko** closed via commit 76ddcb127946e64a94f2608a04fd1b50604cb59f at 2015-04-16 20:03:51_
